### PR TITLE
Fixes #15798: Rationalize the handling of variable during a policy generation

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/15798

### CODE REVIEW

**Comment by ncharles** [comment]
Need to measure the perf impact on this change before merging

**Comment by ncharles** [comment]
the impact is *exactly* 0

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/write/PrepareTemplateVariables.scala [hunk 1]
```diff
@@ -177,46 +169,72 @@ class PrepareTemplateVariablesImpl(
...
-        //to have one directory by directive.
-        val techniqueTemplates = {
-          val templates = allTemplates.filterKeys(k => techniqueTemplatesIds.contains(k._1) && k._2 ==  agentNodeProps.agentType).values.toSet
-          p.technique.generationMode match {
-            case TechniqueGenerationMode.MultipleDirectives =>
-              templates.map( copyInfo => copyInfo.copy(destination = Policy.makeUniqueDest(copyInfo.destination, p)))
-            case _ =>
-              templates
+    for {
+      variableHandler    <- agentRegister.findHandler(agentNodeProps) ?~! s"Error when trying to etch variable escaping method for node ${agentNodeProps.nodeId.value}"
```
Comment (fanf) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/write/PrepareTemplateVariables.scala]:
```suggestion
      variableHandler    <- agentRegister.findHandler(agentNodeProps) ?~! s"Error when trying to fetch variable escaping method for node ${agentNodeProps.nodeId.value}"
```

**Comment by fanf** [comment]
OK, merging this PR

**Pull request merged**